geode-lucene/src/distributedTest/java/org/apache/geode/cache/lucene/internal/IndexRepositoryFactoryDistributedTest.java

```diff
@@ -0,0 +1,209 @@
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.internal.cache.PartitionedRegionDataStore;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+
+@RunWith(JUnitParamsRunner.class)
+public class IndexRepositoryFactoryDistributedTest implements Serializable {
+  private static final String INDEX_NAME = "index";
+  private static final String REGION_NAME = "region";
+  private static final String DEFAULT_FIELD = "text";
+  protected int locatorPort;
+  protected MemberVM locator, dataStore1, dataStore2;
+
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule(5);
+
+  @Before
+  public void setUp() {
+    locator = cluster.startLocatorVM(0);
+    locatorPort = locator.getPort();
+    dataStore1 = cluster.startServerVM(1, locatorPort);
+    dataStore2 = cluster.startServerVM(2, locatorPort);
+  }
+
+  private Cache getCache() {
+    Cache cache = ClusterStartupRule.getCache();
+    assertThat(cache).isNotNull();
+
+    return cache;
+  }
+
+  private void initDataStoreAndLuceneIndex(RegionShortcut regionShortcut) {
+    Cache cache = getCache();
+    LuceneService luceneService = LuceneServiceProvider.get(cache);
+    luceneService.createIndexFactory().setFields(DEFAULT_FIELD).create(INDEX_NAME, REGION_NAME);
+
+    cache.<Integer, TestObject>createRegionFactory(regionShortcut)
+        .setPartitionAttributes(new PartitionAttributesFactory<Integer, TestObject>()
+            .setTotalNumBuckets(1).create())
+        .create(REGION_NAME);
+  }
+
+  private void insertEntries() {
+    Cache cache = getCache();
+    Region<Integer, TestObject> region = cache.getRegion(REGION_NAME);
+    IntStream.range(0, 1000).forEach(i -> region.put(i, new TestObject("hello world" + i)));
+  }
+
+  private void assertPrimariesAndSecondaries(int primaries, int secondaries) {
+    Cache cache = getCache();
+    PartitionedRegionDataStore partitionedRegionDataStore =
+        ((PartitionedRegion) cache.getRegion(REGION_NAME)).getDataStore();
+    assertThat(partitionedRegionDataStore.getAllLocalPrimaryBucketIds().size())
+        .isEqualTo(primaries);
+    assertThat((partitionedRegionDataStore.getAllLocalBucketIds().size()
+        - partitionedRegionDataStore.getAllLocalPrimaryBucketIds().size())).isEqualTo(secondaries);
+  }
+
+  private BucketRegion getFileAndChunkBucket() {
+    Cache cache = getCache();
+    LuceneServiceImpl luceneService = (LuceneServiceImpl) LuceneServiceProvider.get(cache);
+    InternalLuceneIndex index =
+        (InternalLuceneIndex) luceneService.getIndex(INDEX_NAME, REGION_NAME);
+    LuceneIndexForPartitionedRegion indexForPR = (LuceneIndexForPartitionedRegion) index;
+    PartitionedRegion fileRegion = indexForPR.getFileAndChunkRegion();
+
+    return PartitionedRepositoryManager.indexRepositoryFactory.getMatchingBucket(fileRegion, 0);
+  }
+
+  @Test
+  @Parameters({"PARTITION_REDUNDANT"})
```

Comment:
Does this test need to be parameterized if there is only one parameter?

Reply:
Changed! 👍